### PR TITLE
Remove macports from installer and README, and add Installer

### PR DESCRIPTION
**Pull request by EnriqueVidal**
1. ~~Remove hard code paths and aliases that only work for macports.~~
2. ~~Move README command info to it's own README file and subfolder.~~
3. Add installer script to be used by curl

Rel #53

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,19 +46,24 @@ If you already have dotfiles installed (all symlinks have been created) you can
...
-### OSX + macports
+### OSX with [Homebrew](https://github.com/Homebrew/homebrew)
+
+If you're on a mac we highly recommend using [Homebrew](https://github.com/Homebrew/homebrew), it's constantly getting updates and it's pretty clean and easy to manage, here's
+a list of packages you should be looking to install.
 
-* p5-app-ack
-* git-core
+* git
+* git-flow
```
Comment (aalvarado) [inline README.md]: why git-flow?

Comment (EnriqueVidal) [inline README.md]: Why not? noticed the branch name?

Comment (aalvarado) [inline README.md]: ah, I see

**Comment by EnriqueVidal** [comment]
@aalvarado I was thinking moving each part of the project's README to it's sub folder, i.e. the `tmux` part. Into their own folder like we've done with `ncmpcpp` and `vim` and only update those files when we update anything in the project.

Also I want to clean up our two git functions in our bash_profile, right now we're using the long and ugly color characters, back in an older setup I use to have a file with a list of colors as variables and their code. something like this:

```sh
RED="\033[31m"
```
I can't find it anywhere right now, but it'd be a lot cleaner to use those variables in the prompt (or anywhere else) rather than having to code everything up, think you can help me find one?

**Comment by aalvarado** [comment]
@EnriqueVidal will look for one, that will definitely look a lot better. Arch seems to have lots of different bash profiles. They probably have something like that.

**Comment by EnriqueVidal** [comment]
@aalvarado I started an installer in this branch, it should resolve and install only those dependencies needed to generate our dotfiles, i.e. rbenv, git, etc. The idea is to run it like this:

`curl https://raw.githubusercontent.com/cloverinteractive/dotfiles/master/install.sh | sh`

And then it would automatically resolve the dotfiles requirements for Ubuntu, and OSX, it's still WIP but help is greatly appreciated, the script should be very brief and simple, nothing too fancy.

**Comment by EnriqueVidal** [comment]
@aalvarado I added a homebrew installer into the script, however, this means we'd have to handle dependencies differently for every linux flavor, i.e. no `apt-get` in Arch or Gentoo, however [this](https://github.com/Homebrew/linuxbrew) exist. 

Haven't installed it yet, but it looks pretty good, it installs into the home directory so much like this dotfiles every setup is isolated, doesn't require admin rights and does not pollute the system folders, but the best part is, it's exactly the same usage and package names as in homebrew. I was thinking maybe only handling the brew installation differently when the kernel is Darwing or Linux but installing everything else via brew, that should simplify deploying on any machine (until we try on any other Unix flavor that's not Darwin or Linux).

**Comment by aalvarado** [comment]
check out this: https://wiki.archlinux.org/index.php/Color_Bash_Prompt

```
# Reset
Color_Off='\e[0m'       # Text Reset

# Regular Colors
Black='\e[0;30m'        # Black
Red='\e[0;31m'          # Red
Green='\e[0;32m'        # Green
Yellow='\e[0;33m'       # Yellow
Blue='\e[0;34m'         # Blue
Purple='\e[0;35m'       # Purple
Cyan='\e[0;36m'         # Cyan
White='\e[0;37m'        # White

# Bold
BBlack='\e[1;30m'       # Black
BRed='\e[1;31m'         # Red
BGreen='\e[1;32m'       # Green
BYellow='\e[1;33m'      # Yellow
BBlue='\e[1;34m'        # Blue
BPurple='\e[1;35m'      # Purple
BCyan='\e[1;36m'        # Cyan
BWhite='\e[1;37m'       # White

# Underline
UBlack='\e[4;30m'       # Black
URed='\e[4;31m'         # Red
UGreen='\e[4;32m'       # Green
UYellow='\e[4;33m'      # Yellow
UBlue='\e[4;34m'        # Blue
UPurple='\e[4;35m'      # Purple
UCyan='\e[4;36m'        # Cyan
UWhite='\e[4;37m'       # White

# Background
On_Black='\e[40m'       # Black
On_Red='\e[41m'         # Red
On_Green='\e[42m'       # Green
On_Yellow='\e[43m'      # Yellow
On_Blue='\e[44m'        # Blue
On_Purple='\e[45m'      # Purple
On_Cyan='\e[46m'        # Cyan
On_White='\e[47m'       # White

# High Intensity
IBlack='\e[0;90m'       # Black
IRed='\e[0;91m'         # Red
IGreen='\e[0;92m'       # Green
IYellow='\e[0;93m'      # Yellow
IBlue='\e[0;94m'        # Blue
IPurple='\e[0;95m'      # Purple
ICyan='\e[0;96m'        # Cyan
IWhite='\e[0;97m'       # White

# Bold High Intensity
BIBlack='\e[1;90m'      # Black
BIRed='\e[1;91m'        # Red
BIGreen='\e[1;92m'      # Green
BIYellow='\e[1;93m'     # Yellow
BIBlue='\e[1;94m'       # Blue
BIPurple='\e[1;95m'     # Purple
BICyan='\e[1;96m'       # Cyan
BIWhite='\e[1;97m'      # White

# High Intensity backgrounds
On_IBlack='\e[0;100m'   # Black
On_IRed='\e[0;101m'     # Red
On_IGreen='\e[0;102m'   # Green
On_IYellow='\e[0;103m'  # Yellow
On_IBlue='\e[0;104m'    # Blue
On_IPurple='\e[0;105m'  # Purple
On_ICyan='\e[0;106m'    # Cyan
On_IWhite='\e[0;107m'   # White
```

**Comment by aalvarado** [comment]
That sounds like having 2 different installation applications. Let me think about that for a bit.

**Comment by EnriqueVidal** [comment]
314f66e Updates:

* Remove Redhat/Fedora support in favor of keeping things simple (could still be re-introduced by other users if really wanted).
* Add custom branch install support

@aalvarado you can test this via the vagrant setup in the project just:

```bash
vagrant up
vagrant ssh

# This will log you into the virtual machine
/vagrant/install hotfix/house-clean-macports

# Log out and log back in or load ~/.bash_profile
source ~/.bash_profile

# To shutdown vagrant run this
vagrant halt
```

Still need to test this in OSX, I believe the only thing missing for it to build ruby from `rbenv` is `openssl` and `readline`, can you help me test this whilst I merge your other two branch and update the *README* to include installer details, and add the colors in.

maybe @ulisesrmzroche and @marioBonales  could jump in as well? hopefully they read this.

**Comment by aalvarado** [comment]
@EnriqueVidal yes, can do it later today

**Comment by EnriqueVidal** [comment]
@aalvarado not sure if you tested already, but I just checked and there are OSX VirtualBox images available:

https://atlas.hashicorp.com/boxes/search?utf8=%E2%9C%93&sort=&provider=&q=osx

[Here](https://docs.vagrantup.com/v2/boxes.html)'s more info on how to add boxes to vagrant, we can actually test this in a OSX VM as well, without screwing up someone's setup.

Let me know when you run the tests, it'd be great if yo also add this box and run them in a fresh Yosemite or El Capitan box.

**Comment by EnriqueVidal** [comment]
@aalvarado I added a new box with OSX 10.9 with XCode and Command Line tools, this is the way to get them to run:

For linux:

```bash
vagrant linux up
vagrant linux ssh
vagrant linux halt # this halts the vm
```

For OSX:

```bash
vagrant osx up
vagrant osx ssh
vagrant osx halt # this halts the vm
```

If for any reason you need to jump into aqua just run the machine from the VirtualBox App rather than headless from vagrant.

**Comment by aalvarado** [comment]
@EnriqueVidal cool, will check it out later today

**Comment by EnriqueVidal** [comment]
Problem with ruby gems in OSX when running installer, you'll need to get the vbox running, afterwards curl the `install` script from this branch into `/tmp` then add execution permission, afterwards run installer, this problem is only present in OSX, here's how the execution went down for me:

```
vagrant-osx:~ vagrant$ /tmp/install hotfix/house-clean-macports

Installing rbenv for you...
Cloning into '/Users/vagrant/.rbenv'...
remote: Counting objects: 2291, done.
remote: Compressing objects: 100% (52/52), done.
remote: Total 2291 (delta 27), reused 0 (delta 0), pack-reused 2239
Receiving objects: 100% (2291/2291), 401.35 KiB | 30.00 KiB/s, done.
Resolving deltas: 100% (1422/1422), done.
Checking connectivity... done.
rbenv installed
fetching ruby-build...
Cloning into '/Users/vagrant/.rbenv/plugins/ruby-build'...
remote: Counting objects: 5111, done.
remote: Total 5111 (delta 0), reused 0 (delta 0), pack-reused 5111
Receiving objects: 100% (5111/5111), 938.50 KiB | 80.00 KiB/s, done.
Resolving deltas: 100% (2810/2810), done.
Checking connectivity... done.
ruby-build installed
Fetching dotfiles...
Cloning into '/Users/vagrant/.dotfiles'...
remote: Counting objects: 746, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 746 (delta 0), reused 0 (delta 0), pack-reused 743
Receiving objects: 100% (746/746), 313.20 KiB | 39.00 KiB/s, done.
Resolving deltas: 100% (361/361), done.
Checking connectivity... done.
Submodule 'vim/bundle/Vundle.vim' (https://github.com/gmarik/Vundle.vim.git) registered for path 'vim/bundle/Vundle.vim'
Cloning into 'vim/bundle/Vundle.vim'...
remote: Counting objects: 2994, done.
remote: Total 2994 (delta 0), reused 0 (delta 0), pack-reused 2994
Receiving objects: 100% (2994/2994), 892.50 KiB | 106.00 KiB/s, done.
Resolving deltas: 100% (1035/1035), done.
Checking connectivity... done.
Submodule path 'vim/bundle/Vundle.vim': checked out '8db3bcb5921103f0eb6de361c8b25cc03cb350b5'
Installing ruby...
Downloading openssl-1.0.1p.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/bd5ee6803165c0fb60bbecbacacf244f1f90d2aa0d71353af610c29121e9b2f1
Installing openssl-1.0.1p...
Installed openssl-1.0.1p to /Users/vagrant/.rbenv/versions/2.2.3

Downloading ruby-2.2.3.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/df795f2f99860745a416092a4004b016ccf77e8b82dec956b120f18bdc71edce
Installing ruby-2.2.3...
Installed ruby-2.2.3 to /Users/vagrant/.rbenv/versions/2.2.3

ruby installed
Installing dotfiles...
Fetching: bundler-1.10.6.gem (100%)
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.0.0 directory.
./install: line 90: bundle: command not found
```

For some reason after a successful rbenv and ruby installation bash is keeps picking rubygems from the systems:

```
Fetching: bundler-1.10.6.gem (100%)
ERROR:  While executing gem ... (Gem::FilePermissionError)
    You don't have write permissions for the /Library/Ruby/Gems/2.0.0 directory.
./install: line 90: bundle: command not found
```
@aalvarado I really won't be able to work on this for a couple of weeks, can you make sure this gets fixed?

**Comment by aalvarado** [comment]
will try this today now that my image has downloaded

**Comment by aalvarado** [comment]
After several different attempts at getting the images, I've now successfully started the osx vm

**Comment by aalvarado** [comment]
For me it fails when trying to install ruby:

```
Installing ruby...
Usage: rbenv install [-f] [-kvp] <version>
       rbenv install [-f] [-kvp] <definition-file>
       rbenv install -l|--list

  -l/--list          List all available versions
  -f/--force         Install even if the version appears to be installed already
  -s/--skip-existing Skip if the version appears to be installed already

  ruby-build options:

  -k/--keep          Keep source tree in $RBENV_BUILD_ROOT after installation
                     (defaults to $RBENV_ROOT/sources)
  -v/--verbose       Verbose mode: print compilation status to stdout
  -p/--patch         Apply a patch from stdin before building

For detailed information on installing Ruby versions with
ruby-build, including a list of environment variables for adjusting
compilation, see: https://github.com/rbenv/ruby-build#usage
```

**Comment by EnriqueVidal** [comment]
Does your OSX virtual machine has updated command tools? also could be missing a couple of libraries that we need  to include in the installer e.g. openssl

**Comment by aalvarado** [comment]
I'm thinking it was a command line parameter issue. If you do `rbenv install` without the version but if the `.ruby-version` file exists, it will use the version from there correct? I see that you `cd` into the dotfiles directory before running, not sure why it isn't picking it up. 

I did a teardown and up to see if I might have messed something up. 

Will give it another try today

**Comment by EnriqueVidal** [comment]
Bro, you forgot to specify the branch, if you specify this branch it has an .ruby-version file
if you don't then it pulls master and it obviously fail, but yeah that was the problem.

**Comment by aalvarado** [comment]
Ah,
where do I execute this line you mentioned earlier?

> ```
# This will log you into the virtual machine`
/vagrant/install hotfix/house-clean-macports`
```

?

**Comment by EnriqueVidal** [comment]
well, in OSX that's a bit different since there are no VBOX additions you need to do a pull of the current branch (wherever) and just run `./install [branch-name]`

**Comment by aalvarado** [comment]
latest run:

```
vagrant-osx:tmp vagrant$ curl -O https://raw.githubusercontent.com/cloverinteractive/dotfiles/hotfix/house-clean-macports/install
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  4190  100  4190    0     0   4366      0 --:--:-- --:--:-- --:--:--  4364
vagrant-osx:tmp vagrant$ ls
com.apple.launchd.0TEYPfA3kr	com.apple.launchd.aMFb3AmdPh	install
vagrant-osx:tmp vagrant$ chmod +x install 
vagrant-osx:tmp vagrant$ ./install 

Installing rbenv for you...
Cloning into '/Users/vagrant/.rbenv'...
remote: Counting objects: 2325, done.
remote: Compressing objects: 100% (11/11), done.
remote: Total 2325 (delta 3), reused 0 (delta 0), pack-reused 2314
Receiving objects: 100% (2325/2325), 412.58 KiB | 325.00 KiB/s, done.
Resolving deltas: 100% (1448/1448), done.
Checking connectivity... done.
rbenv installed
fetching ruby-build...
Cloning into '/Users/vagrant/.rbenv/plugins/ruby-build'...
remote: Counting objects: 5222, done.
remote: Total 5222 (delta 0), reused 0 (delta 0), pack-reused 5222
Receiving objects: 100% (5222/5222), 973.18 KiB | 476.00 KiB/s, done.
Resolving deltas: 100% (2880/2880), done.
Checking connectivity... done.
ruby-build installed
Fetching dotfiles...
Cloning into '/Users/vagrant/.dotfiles'...
remote: Counting objects: 746, done.
remote: Total 746 (delta 0), reused 0 (delta 0), pack-reused 746
Receiving objects: 100% (746/746), 311.49 KiB | 0 bytes/s, done.
Resolving deltas: 100% (363/363), done.
Checking connectivity... done.
Submodule 'vim/bundle/Vundle.vim' (https://github.com/gmarik/Vundle.vim.git) registered for path 'vim/bundle/Vundle.vim'
Cloning into 'vim/bundle/Vundle.vim'...
remote: Counting objects: 2994, done.
remote: Total 2994 (delta 0), reused 0 (delta 0), pack-reused 2994
Receiving objects: 100% (2994/2994), 892.50 KiB | 591.00 KiB/s, done.
Resolving deltas: 100% (1035/1035), done.
Checking connectivity... done.
Submodule path 'vim/bundle/Vundle.vim': checked out '8db3bcb5921103f0eb6de361c8b25cc03cb350b5'
Installing ruby...
Usage: rbenv install [-f] [-kvp] <version>
       rbenv install [-f] [-kvp] <definition-file>
       rbenv install -l|--list

  -l/--list          List all available versions
  -f/--force         Install even if the version appears to be installed already
  -s/--skip-existing Skip if the version appears to be installed already

  ruby-build options:

  -k/--keep          Keep source tree in $RBENV_BUILD_ROOT after installation
                     (defaults to $RBENV_ROOT/sources)
  -v/--verbose       Verbose mode: print compilation status to stdout
  -p/--patch         Apply a patch from stdin before building

For detailed information on installing Ruby versions with
ruby-build, including a list of environment variables for adjusting
compilation, see: https://github.com/rbenv/ruby-build#usage

vagrant-osx:tmp vagrant$ 
```

still fails on that not seeing `.ruby-version` might be worth checking out if it picks it up by putting that file where the `install` file lives.

**Pull request merged**